apps/envited-x-data-space/common/asset/createTokenMetadata.utils.ts

```diff
@@ -0,0 +1,70 @@
+import { append, equals, path } from 'ramda'
+
+import { TOKEN_TAGS } from '../constants/tokenTags'
+import { extractFilenameFromPath, formatAssetUri, formatIpfsUri } from '../utils'
+import { Manifest } from './types'
+import { formatManifestLinkPath, hasManifestThirdPartyLinks } from './utils'
+
+enum MetadataType {
+  environmentModel = 'environment-model:EnvironmentModel',
+  envitedX = 'envited-x:SimulationAsset',
+  hdmap = 'hdmap:HdMap',
+  ositrace = 'ositrace:OSITrace',
+  scenario = 'scenario:Scenario',
+  surfaceModel = 'surface-model:SurfaceModel',
+}
+
+const GENEREAL_TYPE_DATASOURCE = {
```

Comment:
Surely all these paths from here on down can be generated based on the asset name? They seem to all follow the exact same pattern.